lucet-runtime/lucet-runtime-internals/src/context/mod.rs

```diff
@@ -628,10 +628,12 @@ impl Context {
     ///
     /// TODO: the return type of this function should really be `Result<!, nix::Error>`, but using
     /// `!` as a type like that is currently experimental.
-    #[inline]
-    pub unsafe fn set_from_signal(to: &Context) -> Result<(), nix::Error> {
-        signal::pthread_sigmask(signal::SigmaskHow::SIG_SETMASK, Some(&to.sigset), None)?;
-        Context::set(to)
+    pub unsafe extern "C" fn set_from_signal() -> Result<(), nix::Error> {
```

Comment:
Can we make an `Instance`-specific version of this, and leave the original, as this does something rather different? This is mostly to satisfy my own itch to keep `context` and its submodules cleanly separated from the rest of the codebase.